addons/missileguidance/functions/fnc_onFiredDeffered.sqf

```diff
@@ -0,0 +1,135 @@
+#include "..\script_component.hpp"
+/*
+ * Author: PabstMirror
+ * Gathers onFired args for later use
+ *
+ * Arguments:
+ * 0: Shooter (Man/Vehicle) <OBJECT>
+ * 4: Ammo <STRING>
+ * 6: Projectile <OBJECT>
+ *
+ * Return Value:
+ * Args <Array>
+ *
+ * Example:
+ * [player, "", "", "", "ACE_Javelin_FGM148", "", theMissile] call ace_missileguidance_fnc_onFiredDeffered
+ *
+ * Public: No
+ */
+
+params ["_shooter","_weapon","","_mode","_ammo","","_projectile"];
+
+// Bail if guidance is disabled for this ammo
+if ((getNumber (configFile >> "CfgAmmo" >> _ammo >> QUOTE(ADDON) >> "enabled")) != 2) exitWith { ERROR("not enabled=2"); [] };
+
+// MissileGuidance is enabled for this shot
+TRACE_4("enabled",_shooter,_ammo,_projectile,typeOf _shooter);
+
+private _config = configFile >> "CfgAmmo" >> _ammo >> QUOTE(ADDON);
+
+private _target = _shooter getVariable [QGVAR(target), nil];
+private _targetPos = _shooter getVariable [QGVAR(targetPosition), nil];
+private _seekerType = _shooter getVariable [QGVAR(seekerType), nil];
+private _attackProfile = _shooter getVariable [QGVAR(attackProfile), nil];
+if ((getNumber (configFile >> "CfgAmmo" >> _ammo >> QUOTE(ADDON) >> "useModeForAttackProfile")) == 1) then {
+    _attackProfile = getText (configFile >> "CfgWeapons" >> _weapon >> _mode >> QGVAR(attackProfile))
+};
+private _lockMode = _shooter getVariable [QGVAR(lockMode), nil];
+
+private _laserCode = _shooter getVariable [QEGVAR(laser,code), ACE_DEFAULT_LASER_CODE];
+private _laserInfo = [_laserCode, ACE_DEFAULT_LASER_WAVELENGTH, ACE_DEFAULT_LASER_WAVELENGTH];
+
+TRACE_6("getVars",_target,_targetPos,_seekerType,_attackProfile,_lockMode,_laserCode);
+
+private _launchPos = getPosASL (vehicle _shooter);
+
+if (isNil "_seekerType" || {!(_seekerType in (getArray (_config >> "seekerTypes")))}) then {
+    _seekerType = getText (_config >> "defaultSeekerType");
+};
+if (isNil "_attackProfile" || {!(_attackProfile in (getArray (_config >> "attackProfiles")))}) then {
+    _attackProfile = getText (_config >> "defaultAttackProfile");
+};
+if (isNil "_lockMode" || {!(_lockMode in (getArray (_config >> "seekerLockModes")))}) then {
+    _lockMode = getText (_config >> "defaultSeekerLockMode");
+};
+
+// If we didn't get a target, try to fall back on tab locking
+if (isNil "_target") then {
+    if (!isPlayer _shooter) then {
+        // This was an AI shot, lets still guide it on the AI target
```

Comment:
```suggestion
        // This was an AI shot, let's still guide it on the AI target
```

Reply:
onFiredDeferred is just a copy of ace_missileguidance_fnc_onFired with slight changes to start and end
so I'd really like to just keep it unchanged
to make it easy as possible to migrate it to the missile guidance rework PR